File: fly/src/main/java/com/xia/fly/mvp/BaseMvpPresenter.java
package com.xia.fly.mvp;

import android.content.Context;

import com.xia.fly.mvp.nullobject.MvpNullObjectBasePresenter;
import com.xia.fly.utils.FlyUtils;

import androidx.annotation.CallSuper;
import androidx.annotation.MainThread;
import androidx.annotation.NonNull;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleOwner;

/**
 * @author weixia
 * @date 2019/3/19.
 */
@SuppressWarnings("WeakerAccess")
public abstract class BaseMvpPresenter<V extends BaseMvpView> extends MvpNullObjectBasePresenter<V> {
    protected LifecycleOwner mLifecycleOwner;
    protected Context mContext;

    @CallSuper
    @MainThread
    @Override
    public void onCreate(@NonNull LifecycleOwner owner) {
        mLifecycleOwner = owner;
        mContext = FlyUtils.getContext(owner);
    }

    @CallSuper
    @MainThread
    @Override
    public void onStart(@NonNull LifecycleOwner owner) {
    }

    @CallSuper
    @MainThread
    @Override
    public void onResume(@NonNull LifecycleOwner owner) {
    }

    @CallSuper
    @MainThread
    @Override
    public void onPause(@NonNull LifecycleOwner owner) {
    }

    @CallSuper
    @MainThread
    @Override
    public void onStop(@NonNull LifecycleOwner owner) {
    }

    @CallSuper
    @MainThread
    @Override
    public void onDestroy(@NonNull LifecycleOwner owner) {
    }

    @CallSuper
    @MainThread
    @Override
    public void onLifecycleChanged(@NonNull LifecycleOwner owner, @NonNull Lifecycle.Event event) {
    }
}

File: fly/src/main/java/com/xia/fly/mvp/BaseMvpPresenter.kt
package com.xia.fly.mvp

import android.content.Context

import com.xia.fly.mvp.nullobject.MvpNullObjectBasePresenter
import com.xia.fly.utils.FlyUtils

import androidx.annotation.CallSuper
import androidx.annotation.MainThread
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner

/**
 * @author weixia
 * @date 2019/3/19.
 */
abstract class BaseMvpPresenter<V : BaseMvpView<*>> : MvpNullObjectBasePresenter<V>() {
    protected lateinit var mLifecycleOwner: LifecycleOwner
    protected var mContext: Context? = null

    @CallSuper
    @MainThread
    override fun onCreate(owner: LifecycleOwner) {
        mLifecycleOwner = owner
        mContext = FlyUtils.getContext(owner)
    }

    @CallSuper
    @MainThread
    override fun onStart(owner: LifecycleOwner) {
    }

    @CallSuper
    @MainThread
    override fun onResume(owner: LifecycleOwner) {
    }

    @CallSuper
    @MainThread
    override fun onPause(owner: LifecycleOwner) {
    }

    @CallSuper
    @MainThread
    override fun onStop(owner: LifecycleOwner) {
    }

    @CallSuper
    @MainThread
    override fun onDestroy(owner: LifecycleOwner) {
    }

    @CallSuper
    @MainThread
    override fun onLifecycleChanged(owner: LifecycleOwner, event: Lifecycle.Event) {
    }
}
